doc/documentation_workflow.rst

```diff
@@ -0,0 +1,152 @@
+
+1. You can edit and/or add ``.rst`` files in the ``doc`` directory using your editor of choice.
+
+2. If you create a new page, open ``index.rst`` in the ``doc`` directory and add the file name under ``.. toctree::``. This will ensure it appears on the NEST simulator documentation's table of contents.
+
+3. Save your changes.
+
+Rendering HTML
+~~~~~~~~~~~~~~
+
+Using Sphinx, you can build documentation locally and preview it offline:
+
+1. Go to the ``doc`` folder in the source directory:
+
+.. code-block:: bash
+
+    cd git/nest-simulator/doc
+
+2. Generate HTML files:
+
+.. code-block:: bash
+
+   make html
+
+3. Preview files. They are then located in ``./docs/_build/html``:
+
+.. code-block:: bash
+
+   cd git/nest-simulator/docs/_build/html
+   Firefox FILENAME.html
+
+4. If you add further changes to your files, repeat steps 1-3.
+
+Previewing on Read the Docs
+---------------------------
+
+You can then preview your documentation on your Read the Docs account.
+
+1. Check that unwanted directories are listed in ``.gitignore``:
+
+.. code-block:: bash
+
+   _build
+   _static
+   _templates
+
+2. Add, commit and push your changes to GitHub.
+
+3. Go to `Read the Docs <https://readthedocs.org/>`_. Sign up for an account if you don't have one.
+
+4. `Import <https://readthedocs.org/dashboard/import/>`_ the project.
+
+5. Enter the details of your project in the ``repo`` field and hit ``Create``.
+
+6. `Build your documentation <https://docs.readthedocs.io/en/stable/intro/import-guide.html#building-your-documentation>`_.
+
+This allows you to preview your work on your Read the Docs account. In order to see the changes on the official NEST simulator documentation, please submit a pull request (see below).
+
+Creating pull request
+-------------------
+
+When you feel your documentation work is finished, you can create a pull request to the ``master`` branch of the NEST Source Code Repository. Your pull request will be reviewed by our NEST Documentation Team.
+
+Developer documentation
+-----------------------
+
+For **developer documentation**, we use `doxygen <http://doxygen.org/>`__
+comments extensively throughout NEST.
+After installing NEST, you can extract comments from the source code with
+``make doc`` and a doxygen folder with html files will be generated in the doc
+folder in your source directory.
+
+For a list of commands for SLI and C++, you can access the the online command
+index via the command line
+
+::
+
+   import nest
+   nest.helpdesk()
+
+
+.. note::
+
+ The command ``helpdesk()`` needs to know which browser to launch in order to display
+ the help pages. The browser is set as an option of helpdesk. Please see the file
+ ``~/.nestrc`` for an example setting firefox as browser.
+ Please note that the command helpdesk does not work if you have compiled
+ NEST with MPI support, but you have to enter the address of the helpdesk
+ (file:///install/path/share/doc/nest/index.html) manually into the browser.
```

Comment:
`file:///</path/to/nest_install_dir>/share/doc/nest/index.html`

Reply:
Done.